frontend/js/components/PendingAnswer.js

```diff
@@ -33,7 +35,7 @@ PendingAnswer.propTypes = {
 
 const mapStateToProps = (state, { opponent }) => {
   return {
-    opponent: state.battles.users[opponent],
+    opponent: userSelector(state, opponent),
```

Comment:
Prefer prefixing, for legibility: `selectUser`. You can also add specificity: `selectUserById`